linter: stylish formatter reports wrong source code position

### What version of Oxlint are you using?

0.15.10

### What command did you run?

pnpm dlx oxlint -D no-debugger --format stylish repro.js

### What does your `.oxlintrc.json` config file look like?

Probably none. I run the above command from oxc repo root.


### What happened?

The issue was mentioned in the comment https://github.com/oxc-project/oxc/issues/8422#issuecomment-2605064555, but this didn't seem to get addressed yet.

The code of `repro.js` is:

```js
debugger
```

The command output is a following:

```sh
# NOT OK
$ pnpm dlx oxlint -D no-debugger --format stylish repro.js

/home/hiroshi/code/others/oxc/repro.js
  0:8  error  `debugger` statement is not allowed  eslint(no-debugger)


# OK
$ pnpm dlx oxlint -D no-debugger --format unix repro.js
repro.js:1:1: `debugger` statement is not allowed [Error]


# OK
$ pnpm dlx oxlint -D no-debugger repro.js

  × eslint(no-debugger): `debugger` statement is not allowed
   ╭─[repro.js:1:1]
 1 │ debugger
...


# OK
$ pnpm dlx eslint  --no-config-lookup --rule 'no-debugger: error' --format stylish repro.js

/home/hiroshi/code/others/oxc/repro.js
  1:1  error  Unexpected 'debugger' statement  no-debugger
```

--- a/apps/oxlint/src/output_formatter/mod.rs
+++ b/apps/oxlint/src/output_formatter/mod.rs
@@ -158,10 +158,6 @@ mod test {
         Tester::new().with_cwd(TEST_CWD.into()).test_and_snapshot(args);
     }
 
-    /// disabled for windows
-    /// stylish will output the offset which will be different for windows
-    /// when there are multiple lines (`\r\n` vs `\n`)
-    #[cfg(all(test, not(target_os = "windows")))]
     #[test]
     fn test_output_formatter_diagnostic_stylish() {
         let args = &["--format=stylish", "test.js"];

--- a/apps/oxlint/src/output_formatter/stylish.rs
+++ b/apps/oxlint/src/output_formatter/stylish.rs
@@ -63,9 +63,10 @@ fn format_stylish(diagnostics: &[Error]) -> String {
         };
         let max_len_width = diagnostics
             .iter()
-            .filter_map(|diagnostic| diagnostic.labels())
-            .flat_map(std::iter::Iterator::collect::<Vec<_>>)
-            .map(|label| format!("{}:{}", label.offset(), label.len()).len())
+            .map(|diagnostic| {
+                let start = Info::new(diagnostic).start;
+                format!("{}:{}", start.line, start.column).len()
+            })
             .max()
             .unwrap_or(0);
 
@@ -83,13 +84,12 @@ fn format_stylish(diagnostics: &[Error]) -> String {
                 "\u{1b}[33mwarning\u{1b}[0m"
             };
 
-            if let Some(label) = diagnostic.labels().expect("should have labels").next() {
-                let rule = diagnostic.code().map_or_else(String::new, |code| code.to_string());
-                let position = format!("{}:{}", label.offset(), label.len());
-                output.push_str(
-                    &format!("  \u{1b}[2m{position:max_len_width$}\u{1b}[0m  {severity_str}  {diagnostic}  \u{1b}[2m{rule}\u{1b}[0m\n"),
-                );
-            }
+            let info = Info::new(diagnostic);
+            let rule = diagnostic.code().map_or_else(String::new, |code| code.to_string());
+            let position = format!("{}:{}", info.start.line, info.start.column);
+            output.push_str(
+                &format!("  \u{1b}[2m{position:max_len_width$}\u{1b}[0m  {severity_str}  {diagnostic}  \u{1b}[2m{rule}\u{1b}[0m\n"),
+            );
         }
     }
 

